tools/codegen/dest/lazy_ir.py

```diff
@@ -241,18 +255,32 @@ def this_shape(i: int) -> str:
                 shapes_str = ','.join([this_shape(i) for i in range(returns_length)])
                 meta_out = "std::vector<Shape> shapes{" + shapes_str + "};"
 
-            meta_str = f"""auto out_meta = at::meta::{schema.aten_name}({', '.join(str(a.name) for a in all_args)});
-        {meta_out}"""
+            meta_str = f"""
+            auto out_meta = at::meta::{schema.aten_name}({', '.join(str(a.name) for a in all_args)});
+            {meta_out}"""
         else:
             shape_sig = ComputeShapeSignature(metadata.kernel, func)
             meta_str = f"""
-        auto shapes = {shape_sig.shape_call};"""
+            auto shapes = {shape_sig.shape_call};"""
 
         meta_str += f"""
-        TORCH_INTERNAL_ASSERT(shapes.size() == {returns_length});"""
+            TORCH_INTERNAL_ASSERT(shapes.size() == {returns_length});"""
+
+        symbole_str = aten_symbol(schema)
+        # Rewrite inplace symbol like relu_ into relu
+        # TODO: zero_ is an exception. Need figure out why.
+        if symbole_str == "at::aten::relu_":
```

Comment:
mm, we generally support 'functional' versions of ops directly (see ts_ yaml) - we get inplace versions for free via codegenerated helpers that make e.g. foo_(self,  ...) do something like (self = foo()) using copy.  I am not sure why we support `relu_` operator directly.

https://github.com/pytorch/pytorch/commit/57a60c5a12b80f2173d67901a36b47156fde7023
This was the PR that added relu_ - @alanwaketan do you recall why you didn't just add `relu` here?

Reply:
It sounds reasonable to me to get rid of `relu_()` and just lower `relu()` directly.

I dunno about `relu` specifically, but historically I think part of the reason there were/are some inplace/out= lowerings in XLA is because either the functional variant of the op didn't exist (or the functional variant was composite and defined in terms of the out= variant, although the situation there is better with structured kernels).

I also see a comment about `zero()` - I actually just added a functional `zero()` op, to be used by the functionalization pass. LTC/XLA will also probably want to use it too (main advantage is probably just that `zero` will be more amenable to the LTC codegen) - https://github.com/pytorch/pytorch/pull/75913